src/collection.ts

```diff
@@ -673,7 +673,8 @@ export class Collection<TSchema extends Document = Document> {
         new DropIndexOperation(this as TODO_NODE_3286, '*', resolveOptions(this, options))
       );
       return true;
-    } catch {
+    } catch (error) {
+      if (error instanceof MongoOperationTimeoutError) throw error;
```

Comment:
Do we want to punt this change to the release of Driver 7.0 or are we okay with it being here since the only time this will be seen is if users opt into CSOT?

Reply:
This seems correct, CSOT errors should not be squashed.

Reply:
TODO: Run by team, seems like we shouldn't be squashing any errors here